### DISCUSSION TITLE
SCI-FI Day GitHub Actions integration

### DISCUSSION THREAD

**Discussion by josh-swerdlow** [Green-Software-Foundation/if#653] [General]
Hi,

I got really interested in the GHA integration of IF and wanted to play around with it. As part of SCI-FI day, this use case will be fleshed out already. I was curious if the work is being tracked anywhere and if so, can one participate? The [what's next page ](https://wiki.greensoftware.foundation/whats-next-for-impact-framework) seemed to allude to opening this effort up to the community. If it is not being actively worked on, is there a thread that I could post my progress for others to eventually build off of?

Josh

**Comment by josh-swerdlow** [comment]
@jmcook1186 Hi Joseph, sorry to direct message you, but you were so helpful in responding last time :). 

Is forum the correct place for my question? It hasn't had any traction so wondering if I should put it somewhere else. Maybe the GSF main repo or Hack discussion board?

See also my other question (https://github.com/Green-Software-Foundation/if/discussions/652) which although more verbose, hasn't had traction either :(.

**Comment by jmcook1186** [comment]
Hi @josh-swerdlow sorry for not seeing this - I'm actually not sure what's happening with GHA integrations for sci-fi day - I'm tagging @jawache who might be able to shed some light.

**Comment by jawache** [comment]
Hi @josh-swerdlow sorry for not responding sooner. We're still just in the discussion planning stages right now, we've got some epics we needed to refine before working on this one (they are over in the discussions section if you want to wade in) once those are refined and in the backlog we'll be able to really start thinking about this.

Roughly.

- A GitHub action that computes an impact manifest file and checks it back into the repo and updates a badge based on the aggregate score the manifest file generated.
- A plugin which runs a predefined script (e.g. a benchmark process), captures the utilization and pumps that into the rest of a pipeline that computes the impacts. 

There were some hackathon solutions which if we adjusted slightly might be a starting point for the above.
